tests/drivers/charger/sbs_charger/src/test_sbs_charger.c

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2023 Cirrus Logic, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/charger.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/ztest.h>
+#include <zephyr/ztest_assert.h>
+
+struct sbs_charger_fixture {
+	const struct device *dev;
+	const struct charger_driver_api *api;
+};
+
+static void *sbs_charger_setup(void)
+{
+	static ZTEST_DMEM struct sbs_charger_fixture fixture;
+
+	fixture.dev = DEVICE_DT_GET_ANY(sbs_sbs_charger);
+
+	k_object_access_all_grant(fixture.dev);
+
+	zassert_true(device_is_ready(fixture.dev), "Charger not found");
+
+	return &fixture;
+}
+
+ZTEST_USER_F(sbs_charger, test_get_all_props_failed_returns_negative)
+{
+	struct charger_get_property props[] = {
+		{
+			/* Invalid property */
+			.property_type = CHARGER_PROP_MAX,
+		},
+	};
+
+	int ret = charger_get_prop(fixture->dev, props, ARRAY_SIZE(props));
```

Comment:
suggestion: there's no need to create emulators to test APIs, can be done entirely using FFF.

Reply:
Since we're also testing a driver here, I think using an emulator is a good idea.